asset/asset.go

```diff
@@ -426,7 +426,7 @@ func (w *Witness) Decode(r io.Reader) error {
 }
 
 // DeepEqual returns true if this witness is equal with the given witness.
-func (w *Witness) DeepEqual(o *Witness) bool {
+func (w *Witness) DeepEqual(assetVersion Version, o *Witness) bool {
```

Comment:
Perhaps the tests should be what changes here, or we make a new method for this specific short circuit behavior. I don't think the caller should be forced to know which asset version is used, as the function just seeks to do a direct comparison between the two witness fields.

Reply:
Okay, I refactored this slightly. Not sure about the naming, but is this what you had in mind?